src/main/java/redis/clients/jedis/JedisSafeAuthenticator.java

```diff
@@ -0,0 +1,104 @@
+package redis.clients.jedis;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import redis.clients.authentication.core.SimpleToken;
+import redis.clients.authentication.core.Token;
+import redis.clients.jedis.Protocol.Command;
+import redis.clients.jedis.authentication.JedisAuthenticationException;
+import redis.clients.jedis.exceptions.JedisException;
+import redis.clients.jedis.util.SafeEncoder;
+
+public class JedisSafeAuthenticator {
```

Comment:
Do we have any `public` use case for this?

Reply:
i dont have one

Reply:
changed as protected